src/storage-client/src/controller.rs

```diff
@@ -267,8 +267,14 @@ pub trait StorageController: Debug + Send {
     /// This method is NOT idempotent; It can fail between processing of different
     /// collections and leave the controller in an inconsistent state. It is almost
     /// always wrong to do anything but abort the process on `Err`.
+    ///
+    /// The `register_ts` is used as the initial timestamp that tables are available for reads. (We
+    /// might later give non-tables the same treatment, but hold off on that initially.) Callers
+    /// must provide a Some if any of the collections is a table. A None may be given if none of the
+    /// collections are a table (i.e. all materialized views, sources, etc).
     async fn create_collections(
         &mut self,
+        register_ts: Option<Self::Timestamp>,
```

Comment:
I would really like to see that be part of the `Vec` below, whose only purpose is to present a batch interface and so its inner type is what the "true" arguments of this function would have been if we didn't need batching. This avoids the conundrum of what to pass if you want to create mixed types of collections

Reply:
The issue with that is you then have to handle the situation where the timestamp is different, which feels to me like needless complexity if they'll always be the same ts in practice?

Reply:
Oh, I should say that the way this is structured here is motivated by the upcoming txn stuff. In particular, registration actually requires writing the data shard's id to the txn shard at some timestamp. That makes handling different timestamps for each collection much harder (so the complexity I'm referencing isn't quite what the world looks like now, but what it will look like soon).

That's a pretty minimal overview of the context. Probably easier to catch you up on a huddle with the rest